### PR TITLE
Amusement Park Improvement: Update instruction for `fits_ride?` method

### PR DESCRIPTION
**Pull request by praveendhawan**
The instructions in the readme were unclear. I could not understand how I am supposed to implement it until I saw the argument name in `attendee.rb` file.

### CODE REVIEW

File: exercises/concept/amusement-park-improvements/.docs/instructions.md
Comment (kotp) [inline exercises/concept/amusement-park-improvements/.docs/instructions.md:16]: This seems about right.

One sentence per line, though:

```suggestion
Implement the `Attendee#fits_ride?` method to see if an attendee fits a ride based on their height.
The ride's required minimum height is provided as an argument.
 An attendee must have height greater than or equal to ride's required minimum height.
```

**Review by kotp — CHANGES_REQUESTED** [review]

File: exercises/concept/amusement-park-improvements/.docs/instructions.md
Comment (kotp) [reply exercises/concept/amusement-park-improvements/.docs/instructions.md:16]: The name being in the pre-written part of the solution file is probably
the intended source of truth, rather than the instruction file.

After you opened up the code where you were intended to write, was it
then apparent?

Comment (praveendhawan) [reply exercises/concept/amusement-park-improvements/.docs/instructions.md:16]: After I read the instruction I was confused about what I had to do. When I looked at the source code, I read the name of the argument and it became clear what I had to do. So yes. 

When I saw the instructions of methods in some of the exercises and it made sense to include these instructions. 

I will do the above-proposed changes.

Comment (kotp) [reply exercises/concept/amusement-park-improvements/.docs/instructions.md]: So the remaining question, from a mentor to a student, does it make sense to have the explanation in the README, or have the truth in scaffolding that is given, since "Continuing your work with the amusement park, you are tasked with writing some utility methods to facilitate checking if an attendee can use a ride" and that implies a task of becoming familiar, or presuming that you are already familiar with this code.

Since this comes after the exercise that you have done to complete the _Amusement Park_ exercise, the code that has this is already implemented by you, and arguably, is something that you already know.

I am trying to find out if there is value in having a student make that connection, as I do think you have demonstrated your self that it is not a leap, but a small step to have the question, and then to immediately find it in a place that you have in fact already done, as you have completed the prior exercise, and while we might have chosen a different name in the file that already exists, it probably is very close to your implementation.  The problem solved there has not changed.

Comment (praveendhawan) [reply exercises/concept/amusement-park-improvements/.docs/instructions.md]: As a student learning, I think the clear instructions makes more sense. But it's also true that after seeing the method declaration and arguments,  I understood the assignment.

That's also because I am not new to programming. I could guess what needs to be done and I tried that and the test passed. But doesn't that make it hit and try solution? Having clear instruction in the readme, will allow student to think before even going to the implementation class and maybe think of a way to implement it. 

What do you think?

Comment (LeonardEuler) [reply exercises/concept/amusement-park-improvements/.docs/instructions.md]: This is the difference between concept exercises and practice exercises.  The concept exercises are more geared for "Learning to Program" and the practice exercises are geared for exactly what you stated " that is because I am not new to programming".

The tests for practice exercises are not really meant to be hidden from the student, they are a form of communication and should be used by the student for clarification. 

So you may be right for this as a concept exercise, to maybe write it out.  It would be even better if the students solution for their prior exercise could literally be used for what they (and we) would have for the base of that which is to be improved.

That reminds me, perhaps the pull request title should be updated to indicate the exercise it addresses:

"Amusement Park Improvement: Update instruction for `fits_ride?` method."

**Review by kotp — DISMISSED** [review]

File: exercises/concept/amusement-park-improvements/.docs/instructions.md
Comment (kotp) [reply exercises/concept/amusement-park-improvements/.docs/instructions.md]: @LeonardEuler Thanks for suggesting the improved title for the PR.  I can find it easier in my list of PR's.  And of course thank you @praveendhawan for updating the title.

I will go with whatever you think should be done for the sake of the concept exercise.  I realize that the student may not see the test file if they are using the web interface.

**Comment by kotp** [comment]
@praveendhawan if you have the ability to desire to rebase and merge as is, feel free.  If you would still like to adjust the description, feel free to do so as well.  It should be coming in soon enough, either way if you are happy with it.

**Comment by praveendhawan** [comment]
@kotp I don't have permission to merge. This is my first PR so someone with write access should merge after 2 approvals. I have rebased my branch with the latest main.

File: exercises/concept/amusement-park-improvements/.docs/instructions.md
Comment (iHiD) [inline exercises/concept/amusement-park-improvements/.docs/instructions.md:17]: Does the student know what an "argument" is at this stage in the track?

**Comment by kotp** [comment]
On 22/01/19 11:44AM, Jeremy Walker wrote:
> @iHiD commented on this pull request.
> 
> 
> 
> > @@ -14,6 +14,8 @@ Attendee.new(100).has_pass?
>  ## 2. Check if an attendee fits a ride
>  
>  Implement the `Attendee#fits_ride?` method to see if an attendee fits a ride based on their height.
> +The ride's required minimum height is provided as an argument.
> 
> Does the student know what an "argument" is at this stage in the track?
> 
They should, as they would have also solved the Amusement Park which
takes the arguments.  But I will read the exercise to make sure.

As the instructions for the earlier exercise states:

> # Instructions
> 
> Working with an amusement park, you've been handed a specification to
> design a system to administer attendance and rides. You've been tasked
> with modeling the Attendee (person visiting the park).
> 
> ## 1. Make new attendees
> 
> Implement the `Attendee#initialize` method of the `Attendee` class, it
> should take a height (in centimeters) and store it as an instance
> variable

I think "should take a height" covers it, though not by name.

File: exercises/concept/amusement-park-improvements/.docs/instructions.md
Comment (kotp) [reply exercises/concept/amusement-park-improvements/.docs/instructions.md:17]: Reply by e-mail apparently does not link very well.

Comment (kotp) [reply exercises/concept/amusement-park-improvements/.docs/instructions.md:17]: On 22/01/19 11:44AM, Jeremy Walker wrote:
> @iHiD commented on this pull request.
> 
> 
> 
> > @@ -14,6 +14,8 @@ Attendee.new(100).has_pass?
>  ## 2. Check if an attendee fits a ride
>  
>  Implement the `Attendee#fits_ride?` method to see if an attendee fits a ride based on their height.
> +The ride's required minimum height is provided as an argument.
> 
> Does the student know what an "argument" is at this stage in the track?
> 
They should, as they would have also solved the Amusement Park which
takes the arguments.  But I will read the exercise to make sure.

As the instructions for the earlier exercise states:

> # Instructions
> 
> Working with an amusement park, you've been handed a specification to
> design a system to administer attendance and rides. You've been tasked
> with modeling the Attendee (person visiting the park).
> 
> ## 1. Make new attendees
> 
> Implement the `Attendee#initialize` method of the `Attendee` class, it
> should take a height (in centimeters) and store it as an instance
> variable

I think "should take a height" covers it, though not by name.

**Review by kotp — APPROVED** [review]
After consideration, I think that the change made here in the second exercise of this set (amusement park) that it should also be done in the first, if @praveendhawan wants to do so, that would be great.

**Comment by praveendhawan** [comment]
Hey @kotp is there any way we can connect and discuss what we want to change in the base Amusement Park exercise?

**Pull request merged**
